kosat-core/src/commonMain/kotlin/org/kosat/CDCL.kt

```diff
@@ -389,13 +383,16 @@ class CDCL {
         }
     }
 
+    /**
+     * Reset the solver to the state before the last [solve].
```

Comment:
This comment is quite misleading: you are not resetting _everything" to the state before the last `solve` call. For example, you do not touch phase saving, derived units, var activities, etc.
I would suggest completely removing this method and simply using `cancelUntil(0)` when it is necessary to backtrack to 0th level (on restart or after each solve)